kernel/aster-nix/src/process/signal/mod.rs

```diff
@@ -39,6 +39,12 @@ pub trait SignalContext {
     fn set_arguments(&mut self, sig_num: SigNum, siginfo_addr: usize, ucontext_addr: usize);
 }
 
+pub fn has_pending_signal() -> bool {
```

Comment:
This function should be moved to where it is used. It has only one user, which is `create_new_user_task`. So that is where it should be. And the function should take the current thread as an input to avoid calling `current_thread!` repeatedly since we have found that `current_thread!` is relatively expensive.